### PR TITLE
Add Lead's isAssessment & hasAssessment in EntryFilter

### PR DESCRIPTION
**Pull request by thenav56**
Addresses
- https://github.com/the-deep/server/issues/1395
- https://deep-of.sentry.io/issues/4615292471/?environment=prod&project=1223295&query=is%3Aunresolved+filter&referrer=issue-stream&statsPeriod=30d&stream_index=1

## Changes

Add new filters in EntryFilter
- leadIsAssessment
- leadHasAssessment
- Fix typo

## This PR doesn't introduce any:

- [x] temporary files, auto-generated files or secret keys
- [x] n+1 queries
- [x] flake8 issues
- [x] `print`
- [x] typos
- [x] unwanted comments

## This PR contains valid:

- [x] tests
- [ ] permission checks (tests here too)
- [ ] translations

### CODE REVIEW

File: apps/entry/filter_set.py [hunk 1]
```diff
@@ -643,6 +645,12 @@ def lead_group_label_filter(self, queryset, name, value):
             return queryset.filter(entrygrouplabel__group__title__icontains=value)
         return queryset
 
+    def lead_has_assessment_filter(self, qs, _, value):
+        if value is None:
+            return qs
+        # TODO: We need to add new assessment module filter here after it is deployed
+        return qs.filter(lead__assessment__isnull=not value)
```
Comment (thenav56) [inline apps/entry/filter_set.py]: @Rup-Narayan-Rajbanshi 
We will need to update this to new ARY. Make sure to update this before new assessment merge.

**Review by sudan45 — APPROVED** [review]

**Review by Rup-Narayan-Rajbanshi — COMMENTED** [review]
@subinasr Let me update old assessment  into new assessment_registry in filter.

File: apps/entry/filter_set.py [hunk 2]
```diff
@@ -643,6 +645,11 @@ def lead_group_label_filter(self, queryset, name, value):
             return queryset.filter(entrygrouplabel__group__title__icontains=value)
         return queryset
 
+    def lead_has_assessment_filter(self, qs, _, value):
+        if value is None:
+            return qs
+        return qs.filter(lead__assessmentregistry__isnull=not value)
```
Comment (Rup-Narayan-Rajbanshi) [inline apps/entry/filter_set.py:651]: @thenav56 why not `return qs.filter(lead__assessmentregistry__isnull = value)`

Comment (thenav56) [reply apps/entry/filter_set.py:651]: lead_has_assessment=True
-> lead__assessmentregistry needs to have value so lead__assessmentregistry__isnull=False

lead_has_assessment=False
-> lead__assessmentregistry needs to be empty so lead__assessmentregistry__isnull=True

**Review by Rup-Narayan-Rajbanshi — APPROVED** [review]

**Pull request merged**